pages/api/games/index.js

```diff
@@ -0,0 +1,48 @@
+import { parseString } from "xml2js";
+const baseApiUrl = "https://boardgamegeek.com/xmlapi2/";
+
+// convertor function from library
+const parseXMLToJSON = async (xmlData) => {
+  return new Promise((resolve, reject) => {
+    parseString(
+      xmlData,
+      { explicitArray: false, mergeAttrs: true },
+      (err, result) => {
+        if (err) {
+          console.error("Error in parseString:", err);
+          reject(err);
+        } else {
+          resolve(result);
+        }
+      }
+    );
+  });
+};
+
+// fetch xml from API and converting xml to json
+const fetcher = async (url) => {
+  const response = await fetch(url);
+  const xmlData = await response.text();
+  try {
+    const jsonData = await parseXMLToJSON(xmlData);
+    return jsonData.items.item;
```

Comment:
if you want to stay as generic as possible in this function I would suggest to just return the jsonData instead of the `.items.item`